template/deployments/helm/charts/templates/deployment.yaml

```diff
@@ -1,28 +1,28 @@
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: {{ include "{{APPNAME}}.fullname" . }}
+  name: {{ printf "{{ include \"%s.fullname\" . }}" .APPNAME }}
   labels:
-    {{- include "{{APPNAME}}.labels" . | nindent 4 }}
-    kubernetes.azure.com/generator: {{GENERATORLABEL}}
+    {{ printf "{{- include \"%s.labels\" . | nindent 4 }}" .APPNAME }} {{`
```

Comment:
Its beneficial to have the {{` and `}} be on new lines.  You can probably use something like {{-` and `-}} to remove the whitespace created by them in the result

Reply:
Ahh markdown is making it odd to understand but I mean this only for the comment components and not the ones containing `printf`

Reply:
yeah I understood! I will update it